other/playground/BrjappConnector.ts

```diff
@@ -0,0 +1,224 @@
+            loginFetchResponse.data.errorCode === 'E006'
+        ) {
+            return {
+                isSuccess: false,
+                message: (loginFetchResponse.data?.message || 'Unknown error during login')
+                    .split('Customer')
+                    .join('User'),
+                token: null,
+                isEmailVerificationRequired: false,
+            };
+        } else if (loginFetchResponse.data.errorCode === 'E007') {
+            return {
+                isSuccess: false,
+                message: `You need to verify your email ${email}`,
+                token: null,
+                isEmailVerificationRequired: true,
+            };
+        }
+
+        // Note: User is not registered, try to register
+
+        const registerFetchResponse = await client.POST(`/api/v1/customer/register-account`, {
+            params: {
+                query: {
+                    apiKey,
+                },
+                headers: {
+                    // 'Content-Type': 'application/json',
+                },
+            },
+            body: {
+                email,
+                password,
+                // returnUrl: '',
+                // name: '',
+                // firstName: '',
+                // lastName: '',
+                // phone: '',
+                // companyName: '',
+                // companyRegistrationNumber: '',
+                // taxIdentificationNumber: '',
+                // streetAddress: '',
+                // city: '',
+                // cityPart: '',
+                // stateRegion: '',
+                // postalCode: '',
+                // country: '',
+                newsletter: true,
+                primaryLocale: 'en',
+                groups: this.options.userGroups,
+                customerRealIp,
+                // referralId: '',
+            },
+        });
+
+        if (!(registerFetchResponse.data?.success || false)) {
+            return {
+                isSuccess: false,
+                message: (loginFetchResponse.data?.message || 'Unknown error during registration')
+                    .split('Customer')
+                    .join('User'),
+                token: null,
+                isEmailVerificationRequired: false,
+            };
+        }
+
+        // Note: User is newly registered, add him initial credits
+
+        await this.addInitailCredits(email);
+        // <- TODO: Maybe not await, do it indipendently of return below
+
+        return {
+            isSuccess: true,
+            message: `Registered as ${email}`,
+            token: null,
+            isEmailVerificationRequired: true,
+        };
+    }
+
+    private async addInitailCredits(email: string): Promise<void> {
+        console.log(`Addding initial credits ${this.options.initialCredits} to ${email}`);
+
+        /*
+        TODO: Implement
+        const xxxFetchResponse = await client.POST(`/api/v1/shop/order/create`, {
```

Comment:
Best solution is create a empty order with credit item. And authorize it automatically based on initial status. It should be solved on BRJ side. Please contact me while you solve it.

Reply:
I think here should be special order group for initial extra credit.